### ISSUE TITLE
SSLCertificate Exception thrown on Unrelated Errors

### ISSUE THREAD

**Issue by montymxb** [braintree/braintree_php#173]
In ```Http::_doUrlRequest``` there is a check that can throw an 'ssl' exception based on whether SSL/TLS is on and whether the http status code is 0. Effectively if the request failed and SSL is 'on' this indicates that a certificate issue may be at fault.

This can be misleading if say a "Could not resolve host" (6) error is returned by curl. In this scenario ssl can be enabled but still return a status code of 0, resulting in the sdk throwing an ```SSLCertificate``` exception. A number of underlying issues could be masked by this singular exception besides this one example, so long as SSL is enabled and the request fails.

If it's possible it would be helpful to throw a less vague exception, such as by including any curl error code or message available (if any). This may help to find underlying issues in local setups in a much more straightforward fashion. Furthermore it may be prudent to create and throw a more general exception, something new perhaps like ```CurlError``` or ```ConnectionError``` to fit the general scope of possible exception causes (assuming that ```SSLCertificate``` is not a more appropriate choice for the given error).

If there is an interest to discuss or consider such functionality that would be great. Finally I would be willing to work on a PR if that would help

**Comment by montymxb** [comment]
@EvanHahn just checking to see if there's a general interest in this, in any form. Assuming you're at least one of the right people to contact here?

**Comment by EvanHahn** [comment]
Sorry for the delay!

This sounds like a reasonable request to us. Could you submit a PR? We'd really appreciate it (and appreciate it more if it came with tests).

Also: how did this issue manifest itself in your integration?

**Comment by montymxb** [comment]
@EvanHahn sure I can put up a PR, with tests of course 👍 .

As far as the origin, we observed an issue specifically related to the inability to resolve hosts on our machines. It looks like there was a [bug introduced in glibc](https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1674532) after a standard update. This caused curl via php to attempt to resolve IPv6 over IPv4 by default, without falling back to the later. However all we saw were a series of  `SSLCertificate` exceptions with no details in our logs, which masked the underlying issue. 

It wasn't until we make a quick change to start logging the curl error & code from the sdk that we noticed what was going on. From there we were able to put in a quick patch, just forcing IPv4 resolution until we had the underlying cause fixed. 

Although the issue we observed was unrelated to anything in this sdk it was difficult to pinpoint what was going wrong initially. This brought me to the thought that it would be nice to get a more detailed response in case of bad behavior in any requests. This would help us (and hopefully others) in quickly identifying any issues that could pop up in the future.
